woocommerce/woocommerce
# Prevent sync on read when using HPOS diff CLI tool

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





When using `wp wc hpos diff` to compare an order in its HPOS and posts version, if compatibility mode is enabled and HPOS is authoritative, a "sync on read" might be triggered, thus removing the differences, which is a possibly undesired side effect when comparing orders (precisely because they're different! 😅).

Normally, when compatibility mode is enabled, `wp wc hpos diff` shouldn't provide much value as, in theory, both versions of an order should be identical, but there might be cases where this doesn't happen and this tool should provide a way to investigate the situation without directly changing the results.

This PR makes sure that sync-on-read is not triggered when using the tool.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to **WC > Orders** and create an order (if you don't have any).
1. Take note of the ID of an existing order.
1. Go to **WC > Settings > Advanced > Features** and make sure that HPOS is selected as datastore and that compatibility mode is **enabled**.
   If necessary, first sync orders by running `wp wc hpos sync` in the command line.
1. Run `wp post meta add <order_id> weird_meta_in_posts yes` to add some metadata to only the posts version of the order.
1. Run `wp wc hpos diff <order_id>` to compare both versions. Confirm that the meta you just added (`weird_meta_in_posts`) is listed as a difference.
1. (Optional) Try the above instructions on `trunk`. In the last step, no differences should be listed except, maybe, for the modification date. In any case, a subsequent run of `wp wc hpos diff` would eliminate that difference too.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
